daphne_worker/src/durable/leader_col_job_queue.rs

```diff
@@ -47,21 +47,24 @@ pub struct LeaderCollectionJobQueue {
     #[allow(dead_code)]
     state: State,
     env: Env,
+    touched: bool,
 }
 
 #[durable_object]
 impl DurableObject for LeaderCollectionJobQueue {
     fn new(state: State, env: Env) -> Self {
-        Self { state, env }
+        Self {
+            state,
+            env,
+            touched: false,
+        }
     }
 
     async fn fetch(&mut self, mut req: Request) -> Result<Response> {
-        match (req.path().as_ref(), req.method()) {
-            (DURABLE_DELETE_ALL, Method::Post) => {
-                self.state.storage().delete_all().await?;
-                Response::empty()
-            }
+        let id_hex = self.state.id().to_string();
+        ensure_garbage_collected!(req, self, id_hex, BINDING_DAP_LEADER_COL_JOB_QUEUE);
```

Comment:
this macro expands two things, the end point for delete_all, and the logic for first touch this DO.
For the second task, isn't possible to register the durable object at creation time, rather than at the first touch. Imagine a Setup step, when everything is created and registered.

Reply:
Yeah, it might be possible to stick this in the constructor `new()`, though I'm not sure if this would make a difference performance-wise. FWIW, I'm following a common pattern: https://github.com/cloudflare/workers-rs/blob/main/worker-sandbox/src/counter.rs#L23

Reply:
it is not only related to performance, but to improve the clarity of code.

Reply:
TBH I'm not sure when the constructor is called. Ultimately it's up to the Workers runtime. I think following the example would be a good idea.

Plus, the constructor doesn't return an error, which we would need it to do.